File: go/twoSum.go
package main
import "fmt"

func main(){
	fmt.Println(twoSum([]int{1,2,3}, 5))
}

func twoSum(nums []int, target int) []int {

	var rv []int
	var ht = make(map[int]int)

	for i, n := range nums {
		if val, ok := ht[n]; ok{
			rv = append(rv, val, i)
			break
		}
		ht[target - n] = i

	}
	fmt.Println(ht)
	return rv
}

File: java/twoSum.java
import java.util.*;

public class twoSum {
    static int[] soln(int[] nums, int target) {
        int n = nums.length;
        Map<Integer,Integer> map=new HashMap<>();
        int[] rv = new int[2];
        for (int i = 0; i < n; i++) {
            if (map.containsKey(target - nums[i])) {
                rv[0] = i;
                rv[1] = map.get(target - nums[i]);
                return rv;
            }
            else {
                map.put(nums[i], i);
            }
        }
        return rv;
    }
}




